numpy/core/records.py

```diff
@@ -509,8 +509,8 @@ def __setattr__(self, attr, val):
                     return ret
         try:
             res = fielddict[attr][:2]
-        except (TypeError, KeyError):
-            raise AttributeError("record array has no attribute %s" % attr)
+        except (TypeError, KeyError) as e:
+            raise AttributeError("record array has no attribute %s" % attr) from e
```

Comment:
```suggestion
            raise AttributeError(
                "record array has no attribute %s" % attr
            ) from e
```
Style nit: 80-char line limit